### ISSUE TITLE
Poetry 1.2.0a2: `poetry update` constantly reinstalls dependencies installed from custom repositories

### ISSUE THREAD

**Issue by jleclanche** [python-poetry/poetry#4385]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Arch Linux
- **Poetry version**: 1.2.0a2

## Issue

See behaviour:

![image](https://user-images.githubusercontent.com/235410/129463509-7cc53d27-d8e1-4680-aab2-ce3a28c80d2d.png)

Looking at -vvv output, only thing I can find mentioning the custom package name:
```
gitlab-24844347: 1 packages found for alpaca-api-client >=0.9.2,<0.10.0
   1: fact: alpaca-api-client (0.9.2) depends on pydantic (>=1.8.2,<2.0.0)
   1: fact: alpaca-api-client (0.9.2) depends on requests (>=2.25.1,<3.0.0)
   1: fact: alpaca-api-client (0.9.2) depends on websockets (>=8.1,<9.0)
   1: selecting alpaca-api-client (0.9.2)
```

Encountered this while investigating #3855 which may be related.

**Comment by jamtat** [comment]
+1 for this, similarly running poetry install in a project using a custom repo with a warm cache reinstalls everything unnecessarily

**Comment by radoering** [comment]
Duplicate: #5360
Resolved by: #5362
